tests/test_dataclasses.py

```diff
@@ -259,6 +259,12 @@ def test_serializable_dict(self):
             "seller_id": offer.seller_id,
         }
 
+    def test_from_dict(self):
```

Comment:
Please also add tests of failing examples (e.g. when you pass a dict with wrong keys/values).